packages/kbn-observable/src/observable.test.ts

```diff
@@ -0,0 +1,986 @@
+
+    consoleError.mockRestore();
+  });
+
+  test('can call "next" on observer when no "next" is defined when subscribed', () => {
+    let observer!: SubscriptionObserver<string>;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    source.subscribe();
+
+    expect(observer.next('foo')).toBe(undefined);
+  });
+
+  test('does not call "next" after "complete" is called', () => {
+    let observer!: SubscriptionObserver<string>;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    const values: any[] = [];
+
+    source.subscribe({
+      next() {
+        values.push('next');
+      },
+      complete() {
+        values.push('complete');
+      },
+    });
+
+    observer.next('foo');
+    observer.complete();
+    observer.next('bar');
+
+    expect(values).toEqual(['next', 'complete']);
+  });
+
+  test('subscription is not closed when next throws an error', () => {
+    let observer!: SubscriptionObserver<string>;
+    const cleanup = jest.fn();
+
+    const consoleError = jest.spyOn(console, 'error');
+    consoleError.mockImplementation(noop);
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+      return cleanup;
+    });
+
+    source.subscribe({
+      next() {
+        throw new Error('thrown from next');
+      },
+    });
+
+    observer.next('foo');
+
+    expect(cleanup).not.toHaveBeenCalled();
+    expect(consoleError.mock.calls).toMatchSnapshot('logs');
+
+    consoleError.mockRestore();
+  });
+});
+
+describe('SubscriptionObserver#error', () => {
+  test('input value is forwarded to the observer', () => {
+    let observer!: SubscriptionObserver<string>;
+    const values: any[] = [];
+
+    const source = new Observable<any>(_observer => {
+      observer = _observer;
+    });
+
+    source.subscribe({
+      error(x) {
+        values.push(x);
+      },
+    });
+
+    const err = new Error('foo');
+
+    observer.error(err);
+
+    expect(values).toEqual([err]);
+  });
+
+  test('it does not forward additional arguments', () => {
+    let observer!: SubscriptionObserver<string>;
+    let observerArgs: any;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    source.subscribe({
+      error(...args: any[]) {
+        observerArgs = args;
+      },
+    });
+
+    (observer as any).error('foo', 'bar', 'baz');
+
+    expect(observerArgs).toEqual(['foo']);
+  });
+
+  test('suppresses the value returned from the observer', () => {
+    let observer!: SubscriptionObserver<string>;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    source.subscribe({
+      error(...args: any[]) {
+        return 'bar';
+      },
+    });
+
+    expect(observer.error(new Error('foo'))).toBe(undefined);
+  });
+
+  test('returns undefined when closed', () => {
```

Comment:
Yeah, bad description. Tried to make it a bit clearer. The test is just trying to clarify what happens when you call `error` on an already `error`d (or otherwise closed) observer. Currently we do nothing (e.g. throw or log) and it just returns `undefined`. This is the same as both RxJS and the Observable proposal. But maybe at some point we want to either throw if that happens or error log or something similar. I haven't _really_ thought through the consequences of that though, therefore just following the other impls for now.

Reply:
Nah, in the end decided to remove this test and add another test specifically for `error` being called twice.